src/matrix/distribution.cpp

```diff
@@ -74,6 +74,7 @@ Distribution& Distribution::operator=(Distribution&& rhs) noexcept {
 void Distribution::computeGlobalSize(const LocalElementSize& local_size,
                                      const comm::Size2D& grid_size) noexcept {
   assert(grid_size == comm::Size2D(1, 1));
+  (void) grid_size;
```

Comment:
At first sight this looked strange, than I realized that you need (in release mode) to mark grid_size as used.
I think is better to add a function or macro with a self-explaining name.

Reply:
I agree with you about creating an ad hoc macro, but I'm wondering if we really need to pass this parameter.

Reply:
You are right. Maybe the name can be changed to `computeGlobalSizeForNonDistr`

Reply:
I've removed the superfluous parameter and renamed the member function with the name suggested. I didn't move the assert inside the c'tor, since we would check a value hardcoded few lines above (but in case, I can quickly add it back).